airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/java/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.java

```diff
@@ -65,6 +66,7 @@ public class S3StorageOperations extends BlobStorageOperations {
   private static final String FORMAT_VARIABLE_EPOCH = "${EPOCH}";
   private static final String FORMAT_VARIABLE_UUID = "${UUID}";
   private static final String GZ_FILE_EXTENSION = "gz";
+  private final HashMap<String, AtomicInteger> partCounts = new HashMap<>();
```

Comment:
we're doing multithreaded access... so let's use a ConcurrentHashMap
```suggestion
  private final ConcurrentMap<String, AtomicInteger> partCounts = new HashMap<>();
```

(also, prefer to declare the variable as the interface ConcurrentMap instead of the concrete implementation ConcurrentHashMap)

my bad, I didn't have this in the slack DM because I forgot about it 🤦

Reply:
resolved by 06c6aa439dfb3225b5cfeb60365c8484a36d9fb6